analyzers/src/SonarAnalyzer.CSharp/Rules/PasswordsShouldBeStoredCorrectly.cs

```diff
@@ -0,0 +1,137 @@
+            propertyTracker.MatchProperty(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Identity_PasswordsHasherOptions, "IterationCount")),
+            x => HasFewIterations(x, propertyTracker));
+        Track(
+            propertyTracker,
+            context,
+            "Identity v2 uses only 1000 iterations. Consider changing to identity V3.",
+            propertyTracker.MatchSetter(),
+            propertyTracker.MatchProperty(new MemberDescriptor(KnownType.Microsoft_AspNetCore_Identity_PasswordsHasherOptions, "CompatibilityMode")),
+            x => propertyTracker.AssignedValue(x) is int mode && mode == 0); // PasswordHasherCompatibilityMode.IdentityV2 results to 0
+
+        var argumentTracker = CSharpFacade.Instance.Tracker.Argument;
+        Track(
+            argumentTracker,
+            context,
+            UseMoreIterationsMessageFormat,
+            argumentTracker.MatchArgument(ArgumentDescriptor.MethodInvocation(KnownType.Microsoft_AspNetCore_Cryptography_KeyDerivation_KeyDerivation, "Pbkdf2", "iterationCount", 3)),
+            x => ArgumentLessThan(x, IterationCountThreshold));
+    }
+
+    private static void Rfc2898DeriveBytes(SonarAnalysisContext context)
+    {
+        // Raise when hashAlgorithm is present
+        var argumentTracker = CSharpFacade.Instance.Tracker.Argument;
+        // Exclude the constructors that have a hashAlgorithm parameter
+        var constructorArgument = ArgumentDescriptor.ConstructorInvocation(
+                        ctor => ctor.ContainingType.Is(KnownType.System_Security_Cryptography_Rfc2898DeriveBytes) && ctor.Parameters.Any(x => x.Name == "hashAlgorithm"),
+                        (methodName, comparison) => string.Compare(methodName, "Rfc2898DeriveBytes", comparison) == 0,
+                        null,
+                        x => x.Name == "iterations",
+                        null,
+                        null);
+        var invocationArgument = ArgumentDescriptor.MethodInvocation(KnownType.System_Security_Cryptography_Rfc2898DeriveBytes, "Pbkdf2", "iterations", x => x is 2 or 3);
+        Track(
+            argumentTracker,
+            context,
+            UseMoreIterationsMessageFormat,
+            argumentTracker.Or(
+                argumentTracker.MatchArgument(constructorArgument),
+                argumentTracker.MatchArgument(invocationArgument)),
+            x => ArgumentLessThan(x, IterationCountThreshold));
+
+        // Raise when hashAlgorithm is NOT present
+        var objectCreationTracker = CSharpFacade.Instance.Tracker.ObjectCreation;
+        Track(
+            objectCreationTracker,
+            context,
+            "Use at least 100 000 iterations and a state-of-the-art digest algorithm here.",
+            objectCreationTracker.MatchConstructor(KnownType.System_Security_Cryptography_Rfc2898DeriveBytes),
+            x => x.InvokedConstructorSymbol.Value.Parameters.All(x => x.Name != "hashAlgorithm"));
```

Comment:
Tracking `new Rfc2898DeriveBytes(...)` where an overload constructor WITHOUT `hashAlgorithm` is used.